client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,59 @@ def build_template(
-            create_first_version (bool): create first version of a workfile
-            workfile_creation_enabled (bool): If True, it might create
-                                              first version but ignore
-                                              process if version is created
+            create_first_version (bool): Create first version of a workfile.
+                 When set to True, this option initiates the saving of the
+                 workfile for an initial version. It will skip saving if
+                 a version already exists.
+            workfile_creation_enabled (bool): Whether the call is part of
+                creating a new workfile.
+                When True, we only build if the current file is not
+                an existing saved workfile but a "new" file. Basically when
+                enabled we assume the user tries to load it only into a
+                "New File" (unsaved empty workfile).
+                When False, the default value, we assume we explicitly want to
+                build the template in our current scene regardless of current
+                scene state.
 
         """
-        if any(
-            value is None
-            for value in [
-                template_path,
-                keep_placeholders,
-                create_first_version,
-            ]
-        ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
-            if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
-            if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
 
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
-
-        # if first version is created, import template
-        # and populate placeholders
+        # Get default values if not provided
         if (
-            create_first_version
-            and workfile_creation_enabled
-            and created_version_workfile
+            template_path is None
+            or keep_placeholders is None
+            or create_first_version is None
         ):
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
+            if keep_placeholders is None:
+                keep_placeholders: bool = preset["keep_placeholder"]
+            if create_first_version is None:
+                create_first_version: bool = preset["create_first_version"]
+
+        # Build the template if we are explicitly requesting it or if it's
+        # an unsaved "new file".
+        is_new_file = not self.host.get_current_workfile()
+        explicit_build_requested = not workfile_creation_enabled
```

Comment:
Creating negated variable does help how?

Reply:
Ok, if that is the case, we should change the argument name too.

From what I've looked for it is used only in nuke, so I suggest to add `explicit_build_requested` to args, keep `workfile_creation_enabled` there but default value is `None`. If `workfile_creation_enabled` won't be `None` we will warn about deprecation and use the value for `explicit_build_requested`.

Reply:
@MustafaJafar can you address this comment?

Reply:
Honestly - if we're refactoring. I'd change the system around that `build_template` always builds, regardless and that we need OTHER logic handling the case that defines whether or not we want to trigger it in a new file, etc. It's just weird functionality for a `build_template` function?

Reply:
That is change of functionality, not better argument names and more separated functionality.

Not saying "no", but we should maybe find out why it was added at first place, and why?

Reply:
My point is that the method expects `workfile_creation_enabled` which is used only to convert it to different variable -> we should change the arguments of the method, with backwards compatibility, if the argument is wrong.

Reply:
And my point is - that if we are renaming the argument now; we're wasting time, because if we do - we should refactor the full `build_template` to just always build... and have the decision made on "whether it's a new file and if it should be saved or not" to be made outside of it". So that calling `build_template` does just that, build it. I'd even argue that because it also **loads** a template that it's a bit ambiguous with the name.

Or at least clean it up so that it's doing something more understandable. :D

Reply:
I don't think we'll get to refactor of this code in next few months. Unless it is done by community, it won't change for a long time.

If you insist on not doing it here, then move this line to first line of function and add comment above it like
```
# More accurate variable name
# - logic related to workfile creation should be moved out in future
```

Reply:
You've reacted with emoji, but nothing changed?

Reply:
https://github.com/ynput/ayon-core/pull/871/commits/7298ddb745e003ca8e317eae3baccab246a2ad37